Firmware/AVR/AVR_DRIVER/src/main.c

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file main.c
+ * @author Tsugumi Murata (tmurata293)
+ * @date 17 Feb 2021
+ * @brief Main File for AVR DRIVER
+ *
+ */
+
+#include <avr/io.h>
+#include <avr/interrupt.h>    
+#include <util/delay.h>      
+#include <stdint.h>                 
+#include "usiTwiSlave.h"            
+#include "pinConfig.h"
+#include "motor.h"
+#include "encoder.h"
+#include "left_driver_peripherals.h"
+#include "right_driver_peripherals.h"
+#include "decodeI2C.h"
+
+static uint8_t getDriverMode(){
+    // check which driver avr it is 
+    DDRB &= ~_BV(MODE_SELECT); 
+    return (PINB & _BV(MODE_SELECT));
+}
+
+
+int main(void)
+{
+    // set system clock to 8MHz
+    CLKPR  = _BV(CLKPCE);
+    CLKPR  = 0;
+    
+    // define local variables 
+    uint8_t driver_mode = 0; 
+    uint8_t slave_address = 0x00;  
+    uint8_t DATA_TO_MASTER_STATUS_REGISTER  = 0; 
+
+    //check mode pin 
+    driver_mode = getDriverMode(); 
+
+    //config for left driver 
+    if (driver_mode) {
+        slave_address = 0x0A;   //slave address for left driver avr 
+        setup_TOF_XSHUT_Config();
+        
+    }
+    //config for right driver 
+    else{
+        slave_address = 0x2A ;   //slave address for right driver avr 
+        //setup only needed for right avr driver 
+        setupWaterLevelConfig();
+        setupMistActuatorConfig();
+    }
+
+    //setup usi twi settings 
+    setupUsiTwiConfig();
+
+    //setup encoder interrupt settings 
+    setupEncoderConfig(); 
+    //setup motor config 
+    setupMotorConfig(PHASE_CORRECT_PWM); 
+
+
+    //initialize the USI communicatin
+    usiTwiSlaveInit(slave_address);
+
+    while(1)
+    { 
+        // reset DATA_TO_MASTER_STATUS_REGISTER to 0 
+        DATA_TO_MASTER_STATUS_REGISTER  = 0;
+
+        //if data received from master
+        if(usiTwiDataInReceiveBuffer()){
+
+            // store the first byte of data 
+            char message_first_byte = usiTwiReceiveByte();
+
+            // check if data is indeed the first data byte of message 
+            if (checkDataHeader(message_first_byte, FIRST)){
+                
+                // check if eStop bit is enabled 
+                if (message_first_byte & ESTOP_COMMAND_REQ_MASK)    eStopMotor();
+
+                //resume if eStop bit is not enabled 
+                else{
+
+                    //common for both driver 
+                    if (message_first_byte & ENCODER_DATA_REQ_MASK) DATA_TO_MASTER_STATUS_REGISTER |= _BV(SEND_ENCODER_DATA); 
+
+                    // decode for left driver 
+                    if(driver_mode){
```

Comment:
a good practice to have a common header file, that stores these enums.

Like tof_config can be enum

Reply:
AVR DRIVER COMMON HEADER FILE V1.1 created